iothub/device/src/TransportSettings/IotHubClientHttpSettings.cs

```diff
@@ -21,6 +21,21 @@ public IotHubClientHttpSettings()
             Protocol = IotHubClientTransportProtocol.WebSocket;
         }
 
+        /// <summary>
+        /// The HTTP client to use for all HTTP operations.
+        /// </summary>
+        /// <remarks>
+        /// If not provided, an HTTP client will be created for you based on the other settings provided.
+        /// <para>
+        /// This HTTP client instance will be disposed when the device/module client using it is disposed.
```

Comment:
The disposal code seems to be missing from this PR. Was it already done as a part of the HttpTransportHandler implementation for the HttpClient that the sdk was creating and now we've substituted the user supplied client for the appropriate case?

Reply:
Correct. ```HttpClientHelper``` was already disposing of the local HttpClient field. This PR just changes how it is instantiated/not instantiated